Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/SimplePlayer.java

```diff
@@ -0,0 +1,109 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class SimplePlayer implements Player {
+
+	private boolean isNew=false;
+	private String name;
+	private Score mScore;
+	private final String DEFAULT_PATH="players.txt";
+
+	//--------------------------------------------
+	
+	public SimplePlayer (String nName){
+		if(!this.loadFromFile( DEFAULT_PATH,nName)){
+		isNew=true;
+		name = nName;
+		mScore = new SimpleScore();
+		}
+		
+	}
+	
+	public void writeToFile(){
+		writeToFile (DEFAULT_PATH);
+	}
+
+	public void writeToFile(String fileName){
+		try (FileWriter w = new FileWriter (fileName);){
+			w.write(name+"\n");
+			w.flush();
+			w.write(mScore.toString());
+			w.flush();
+		} catch (IOException e) {
+			e.printStackTrace();//Shouldn't happen
+		}
+
+	}
+	
+	public boolean loadFromFile(String fileName, String playerName){
+		try(BufferedReader br = new BufferedReader (new FileReader(fileName))){
+			String temp= " ";
+			while (null!=temp){
+				temp = br.readLine();
+				if(temp!=null && temp.equals(playerName))
+				{
+					name = temp;
+					temp = br.readLine();
+					if(temp!=null){
+					mScore = new SimpleScore (temp);
+					}else{mScore = new SimpleScore();}
+					return true;
+				}
+			}
+			return false;
+		} catch (FileNotFoundException e) {
+			File file = new File (DEFAULT_PATH);
+			//Try to create the Default File, just in case
+			try {
+				file.createNewFile();
+			} catch (IOException e1) {
+				e1.printStackTrace();
+			}
+			e.printStackTrace();
+			return false;
```

Comment:
I think you should just keep bubbling the exception up, this is a very reasonable method for an IOException to be thrown from and callers may be interested in it. Also, if a caller explicitly asked not to use DEFAULT_PATH by specifying a path, they probably don't want to use the default path. If they want to use DEFAULT_PATH as a fallback, leave that to them. So, personally, I would not catch the exception here and would not try a second attempt.